components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/core/DropboxAPIFacade.java

```diff
@@ -125,18 +136,18 @@ private DropboxFileUploadResult putFile(String localPath, DropboxUploadMode mode
             }
             return result;
         } else if (fileLocalPath.isDirectory()) {
-            //verify uploading of a list of files inside a dir
+            // verify uploading of a list of files inside a dir
             LOG.debug("Uploading a dir...");
-            //check if dropbox folder exists
-            if (entry != null && !entry.isFolder()) {
+            // check if dropbox folder exists
```

Comment:
Should likely be removed as we dont know if the entry is folder or file

Reply:
As i understand, this can be removed, completely.

 if (entry != null) {
                throw new DropboxException(dropboxPath + " exists on dropbox and is not a folder!");
            }